### PR TITLE
Add environments/coreos.mk

### PR DESCRIPTION
**Pull request by schu**

### CODE REVIEW

File: environments/coreos.mk [hunk 1]
```diff
@@ -0,0 +1,42 @@
+include build.mk
+
+DISTRO=coreos
```
Comment (alban) [inline environments/coreos.mk]: I am ignorant of the "Make" tool details: how is the variable DISTRO available in build.mk if it is declared after the include that uses it?

File: circle.yml [hunk 2]
```diff
@@ -14,7 +14,9 @@ machine:
 dependencies:
   override:
     - |
-      cd $SRCDIR && make fedora-24 && make arch && make debian-testing && ./tools/export-elfs-into-container
+      cd $SRCDIR && make fedora-24 && make arch && make debian-testing
+      test $CIRCLECI == "true" || make coreos
```
Comment (alban) [inline circle.yml]: missing quotes?

```
test "$CIRCLECI" == "true" || make coreos
```

File: environments/coreos.mk [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+build: fetch-image mount-image extract-header umount-image build-docker-image build-ebpf-object cleanup
+
+fetch-image:
+	@echo $(DIR)
+	@curl -L https://$(RELEASE_CHANNEL).release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 | bzip2 -d > $(DIR)/coreos_developer_container.bin
+
+mount-image:
+	@mkdir -p $(DIR)/mnt
+	@sudo mount -o ro,loop,offset=$$((4096*512)) $(DIR)/coreos_developer_container.bin $(DIR)/mnt
```
Comment (alban) [inline environments/coreos.mk]: add a comment to explain that 4096 comes from the first sector, visible with fdisk -l

File: environments/coreos.mk (hunk 1 above)
Comment (schu) [inline environments/coreos.mk]: AFAIK make doesn't care: "Most variable names are considered to have the empty string as a value if you have never set them."

https://www.gnu.org/software/make/manual/html_node/Setting.html#Setting

File: environments/coreos.mk (hunk 3 above)
Comment (schu) [inline environments/coreos.mk]: Done.

File: environments/coreos.mk (hunk 1 above)
Comment (schu) [inline environments/coreos.mk]: And the target gets only executed after we set the var.

**Comment by schu** [comment]
Tested manually on CoreOS alpha (with Linux `4.8.0-1-amd64`).

**Comment by schu** [comment]
We now have both `RELEASE` and `RELEASE_CHANNEL` to build for different versions:

```
RELEASE?=current
RELEASE_CHANNEL?=alpha
```

**Comment by alban** [comment]
> We now have both `RELEASE` and `RELEASE_CHANNEL` to build for different versions:

@schu but only one of them gets included in the ebpf.tgz. Could you make the CI build all of them in the Docker image? SemaphoreCI says:

```
ebpf/
ebpf/fedora/
ebpf/fedora/x86_64/
ebpf/fedora/x86_64/4.8.11-200.fc24.x86_64/
ebpf/fedora/x86_64/4.8.11-200.fc24.x86_64/ebpf.o
ebpf/arch/
ebpf/arch/x86_64/
ebpf/arch/x86_64/4.8.12-2-ARCH/
ebpf/arch/x86_64/4.8.12-2-ARCH/ebpf.o
ebpf/coreos/
ebpf/coreos/x86_64/
ebpf/coreos/x86_64/4.8.11-coreos-r1/
ebpf/coreos/x86_64/4.8.11-coreos-r1/ebpf.o
ebpf/debian/
ebpf/debian/x86_64/
ebpf/debian/x86_64/4.8.0-1-amd64/
ebpf/debian/x86_64/4.8.0-1-amd64/ebpf.o
```

**Comment by alban** [comment]
https://github.com/kinvolk/tcptracer-bpf/pull/9#issuecomment-265213687 can be done in a subsequent PR. LGTM.

**Comment by alban** [comment]
@schu oops, the build on semaphore [passed before](https://semaphoreci.com/alban/tcptracer-bpf/branches/schu-coreos/builds/6) but after the merge, [it fails](https://semaphoreci.com/alban/tcptracer-bpf/branches/master/builds/7) with:
```
No space left on device
```

**Pull request merged**